chia/full_node/mempool_manager.py

```diff
@@ -118,19 +119,25 @@ def compute_assert_height(
 
 
 def run_for_cost_and_additions(
-    coin_id: bytes32, puzzle_reveal: SerializedProgram, solution: SerializedProgram, max_cost: uint64
+    coin_id: bytes32,
+    puzzle_reveal: SerializedProgram,
+    solution: SerializedProgram,
+    max_cost: uint64,
+    npc_result: NPCResult,
 ) -> Tuple[uint64, List[Coin]]:
-    cost, conditions_program = puzzle_reveal.run_mempool_with_cost(max_cost, solution)
+    if npc_result.conds is None:
+        raise ValueError("Invalid NPCResult")
     created_coins = []
-    for s in conditions_program.as_iter():
-        atoms = s.as_atom_list()
-        if len(atoms) == 0:
-            raise ValueError("Invalid condition while running for cost and additions")
-        if atoms[0] == ConditionOpcode.CREATE_COIN:
-            puzzle_hash = bytes32(atoms[1])
-            amount = uint64(int_from_bytes(atoms[2]))
-            created_coins.append(Coin(coin_id, puzzle_hash, amount))
-    return (uint64(cost), created_coins)
+    for spend in npc_result.conds.spends:
+        if spend.coin_id == coin_id:
```

Comment:
you'll only ever find a single spend matching the coin ID, you don't need to keep searching once you've found it.
I think a more clear way of saying this is to first find that spend, then do the loop you have below, to copy the created coins.

iirc, there's an `index()` function in python that returns the index to a matching element in a list. I'm not sure that would work here. But something like that I think would make sense.

Reply:
Done.